### ISSUE TITLE
ASB 06/2017

### ISSUE THREAD

**Issue by MSe1969** [derfelot/discussion#7]
ASB 06/2017 has been published, however - not yet with links to the patches...

**Comment by MSe1969** [comment]
Meanwhile, Google has updated the links to the AOSP changes, so I will tonight work through it and create a tracking list and also start working on the changes.

**Comment by MSe1969** [comment]
Hi all,

the June ASB affects the following components:

packages/apps/Bluetooth	
external/bluetooth/bluedroid
external/libhevc   	
external/libvpx		
external/libxml2
kernel/sony/msm8974
system/core		

@derfelot - we need a fork of **LineageOS/android_external_libxml2** 
For the others, I'll create a cm-12.1-test branch and update the local manifest in 'cm-12.1-test' accordingly later today.

**Comment by MSe1969** [comment]
Btw - as the work to get our kernel to 3.4.113 will require a lot of testing and work (where I unfortunately can only support by means of testing with my device), I'll continue working on our 'current' kernel for the June ASB.

**Comment by Micha-Btz** [comment]
@MSe1969 do you will patch all of these CVE or should we split the work?

The kernel is a problem itself. Maybe it would be better to wait if Jonas has success with boot and running. Or we should create a new branch for the latest kernel patch level like Jonas did.

**Comment by hahnjo** [comment]
There are two vulnerabilities in the kernel, but I don't yet see Google's patches. Can you give a link if you've found them?

If needed, I can do the merge afterwards. I *hope* that this shouldn't be that particularly hard.

**Comment by Micha-Btz** [comment]
I don't have take a look at the June CVE for the kernel now but normally I take a look here.
https://review.lineageos.org/#/q/p:%255E.*8974.*

**Comment by MSe1969** [comment]
@Micha-Btz - I'll do it now. 
Normally, I create the test branches as follows (assumption - cm-12.1 is up to date, otherwise `git pull`):
```
git checkout cm-12.1
git branch cm-12.1-test
git checkout cm-12.1-test
git merge cm-12.1
git push -u origin cm-12.1-test
```

**Comment by MSe1969** [comment]
Okay - I have now done the following:

**Created 'cm-12.1-test' branches for below repo's and updated the local manifest in 'cm-12.1-test':**
packages/apps/Bluetooth
external/bluetooth/bluedroid
external/libhevc
external/libvpx
system/core

**Waiting for fork of external/libxml2**
I can alternatively create the fork in my space and later, @derfelot will fork from my repo?

**Created repo for 'Privacy-Browser' and updated local manifest in 'cm-12.1-test'**


What I am however not yet sure about is how we deal with the **kernel**:
- Should we already ship the updated kernel 3.4.113 together with ASB 06/2017?
- Or should we treat that separately? (i.e. create another branch where we only apply the June patches, and publish a further build with the new kernel to have more time for testing etc.)

Please advise, thanks.

I will not touch the kernel, and simply start patching the other stuff - will first create an overview sheet again and then we all can start contributing.

**Comment by hahnjo** [comment]
@MSe1969 depends on how fast the build shall be published. IMO if we don't find a problem with 3.4.113 in a week or so, a possible issue probably doesn't trigger with our usage scenario. In that case, it shouldn't make much of a difference if we wait an additional month. Just my 2 cents...

**Comment by MSe1969** [comment]
@hahnjo - I leave it up to you and @Micha-Btz to decide.

A proposal from my side:
Independent from the work distribution, we should always have two strictly separated testing scenario's at the beginning:
- a build 'cm-12.1-test' with the patches (except kernel) on the 'cm-12.1' stable branch kernel
- a build 'cm-12.1' without the patches on the new 3.4.113 kernel (cm-12.1-test)

By doing so, we can avoid having to wonder in case of a testing issue, whether it is patching or kernel related. If both test scenarios look good, we merge and test 'cm-12.1-test' = patches+kernel

Does that sound ok?
What do you think?

**Comment by hahnjo** [comment]
@MSe1969 this needs 3 test iterations. To avoid that, we should better do the patching and release a build with that. Afterwards, we can test 3.4.113 independently.

**Comment by Micha-Btz** [comment]
@MSe1969 I have merged the kernel patches 3.4.113 to test so we should create a new branch for this june patches. So that we can make more long term tests. 
And just do the needed fork like that one for V8.

I will make a list for patches we need for the kernel for June and will create a new branch.

**Comment by derfelot** [comment]
I've made a fork and invited all you guys

**Comment by Micha-Btz** [comment]
- CVE-2017-7371 	file not available
- CVE-2017-7365 	file not available
- CVE-2017-7366 	GPU driver maybe backport needed
- CVE-2017-7367 	file not available
- CVE-2016-5861 	file not available
- CVE-2016-5864 	file not available
- CVE-2017-6421 	file not available
- CVE-2017-7364 	file not available
- CVE-2017-7368 	file not available
- CVE-2017-7369 	file not available
- CVE-2017-7370 	file not available
- CVE-2017-7372 	file not available
- CVE-2017-7373 	file not available
- CVE-2017-8233 	Camera code doesn't fit - camera per wrapper from jelly bean
- CVE-2017-8234 	Camera code doesn't fit - camera per wrapper from jelly bean
- CVE-2017-8235 	Camera code doesn't fit - camera per wrapper from jelly bean
- CVE-2017-8236 	file not available
- CVE-2017-8237 	file not available
- CVE-2017-8242 	code doesn't fit, unsure if we are affected
- CVE-2017-8239 	Camera code doesn't fit - camera per wrapper from jelly bean
- CVE-2017-8240 	file not available
- CVE-2017-8241 	file not available

Here is the list of June kernel CVE. Would be great if someone can have a second look at 
CVE-2017-7366 and CVE-2017-8242 for this two I'm unsure.

**Comment by MSe1969** [comment]
I've created a tracking file for the 2017-06-01 patches ("core android" w/o kernel) and put it into the discussion repo. From the five patches marked as "not publicly available" (new Google game? wtf!), I have found two and will implement - the other three, I have given up.

**Comment by MSe1969** [comment]
the two CVE's, which I found despite 'not publicly available' concern external/pdfium - @derfelot we need a fork here, too - from android.googlesource.com/platform/external/pdfium

**Comment by MSe1969** [comment]
OK, done the 2017-06-01 patches, except the two for pdfium.
Test build running . . .

**Comment by hahnjo** [comment]
@Micha-Btz thanks for collecting this list of CVEs!

The patch for CVE-2017-7366 only required minimal conflict resolution, I think we *are* affected. I will need to take a closer at the two patches for CVE-2017-8242, though. If they are applicable, at least the first one will need quite a bit of manual work that I don't have time for just now.

May I push the committed fix for CVE-2017-7366 that I have locally and override your commit to "fix" CVE-2016-1583? As written in #5 this doesn't compile and isn't needed anyway.

**Comment by Micha-Btz** [comment]
I have revert the commit so you can start at a clean tree.

**Comment by hahnjo** [comment]
I simply force-pushed. Please not the commit has only been compile-tested for now! I will look after CVE-2017-8242 ASAP

**Comment by MSe1969** [comment]
Hi, the 2017-06-01 fixes are now all implemented, I forked platform/external/pdfium to my repo.
@derfelot can later fork into here.

**Comment by hahnjo** [comment]
Oops, I mixed up the CVEs in my previous comments: CVE-2017-8242 is done, I have to look after CVE-2017-7366!

**Comment by hahnjo** [comment]
@Micha-Btz Regarding CVE-2017-7366: The GPU driver has changed a lot and most of the functions simply don't exist in our 3.4 kernel. I don't see a reliable way to backport the two commits and give up on that for now.

For 2016-06-05, there is CVE-2015-7995 which affects libxslt. I haven't found a patch though, is that even used with Android 5?

**Comment by MSe1969** [comment]
@hahnjo : CVE-2015-7995 is listed as only applicable for 4.4.4, so I assume not

To all - is my understanding correct, that regarding the 2017-06-05 section, only CVE-2017-7366 and CVE-2017-8242 is relevant? CVE-2017-8242 already done and CVE-2017-7366 deferred or even n/a?

In that case, we should build, test and ship...?!

**Comment by hahnjo** [comment]
@MSe1969 Yes :+1: - as long as Google doesn't release patches for CVE-2017-0648 and CVE-2017-0651 which are listed under the general kernel section.

**Comment by MSe1969** [comment]
Doing clean build right now - will ship then, and merge stuff into cm-12.1 and remove the 'cm-12.1-test' branches (of course - EXCEPT kernel :) !! ) - @derfelot still needs to fork my 'pdfium' repo

**Comment by Micha-Btz** [comment]
great. I have also not found patches for CVE-2017-0648 and CVE-2017-0651. 
Maybe there are patches for this available later. Then we are done with June.

**Comment by hahnjo** [comment]
@MSe1969 Will you merge `cm-12.1-test`? I've already merged `cm-12.1-june` for the kernel...

**Comment by MSe1969** [comment]
Yes, will do - have waited for the fork by @derfelot , but better do now and close this issue.

**Comment by derfelot** [comment]
Sorry, you guys are too quick for me :p 

Thought I already forked everything. Done now. Should be mroe responsive again starting next week :)

**Comment by MSe1969** [comment]
Hi all,

I have done now the following - EXCEPT for our kernel repo, which I have left **untouched**:

1. Merged all 'cm-12.1-test' branches into 'cm-12.1'
2. Verified for each affected repository, that 'cm-12.1' is even with 'cm-12.1-test'
3. Updated local manifest in 'cm-12.1': added removals and additions from cm-12.1-test manually
4. Updated local manifest in 'cm-12.1-test': Reverted all 'cm-12.1-test' references into 'cm-12.1', with the exception of 'build', which by definition always points to 'cm-12.1-test'. Further, I have pointed the kernel to 'cm-12.1-test', as 3.4.113 is now our next thing to do
5. deleted all 'cm-12.1-test' branches - except for kernel and build

Final note: 
Soren Stoutner has provided a repo to include the browser as pre-built, so pointing to his git instead of mine (which I am going to delete eventually)

Having said so - I am now closing this issue and announce the merge in XDA for those wanting to build themselves.
